tsdb/chunks/head_chunks.go

```diff
@@ -539,9 +539,7 @@ func (cdm *ChunkDiskMapper) IterateAllChunks(f func(seriesRef, chunkRef uint64,
 	defer cdm.writePathMtx.Unlock()
 
 	defer func() {
-		if err == nil {
-			cdm.fileMaxtSet = true
-		}
+		cdm.fileMaxtSet = true
```

Comment:
Can you elaborate why this is changed? Not sure

Reply:
We only want to make sure that `IterateAllChunks` was called at least once. Errors are fine here as the `Head` recovers from it (either calls repair on this or skips m-mapped chunks as there is data in WAL).

Head skipping m-map chunks and going with WAL is a possibility in case of non-corruption errors and we should be able to run compaction in that case. `Truncate()` on m-mapped chunk fails during compaction if this is not set to true in case of error.